mycroft/util/log.py

```diff
@@ -126,18 +126,23 @@ def _log(cls, func, *args, **kwargs):
             # [0] - _log()
             # [1] - debug(), info(), warning(), or error()
             # [2] - caller
-            stack = inspect.stack()
-
-            # Record:
-            # [0] - frame object
-            # [1] - filename
-            # [2] - line number
-            # [3] - function
-            # ...
-            record = stack[2]
-            mod = inspect.getmodule(record[0])
-            module_name = mod.__name__ if mod else ''
-            name = module_name + ':' + record[3] + ':' + str(record[2])
+            try:
+                stack = inspect.stack()
+
+                # Record:
+                # [0] - frame object
+                # [1] - filename
+                # [2] - line number
+                # [3] - function
+                # ...
+                record = stack[2]
+                mod = inspect.getmodule(record[0])
+                module_name = mod.__name__ if mod else ''
+                name = module_name + ':' + record[3] + ':' + str(record[2])
+            except Exception:
+                # The location couldn't be determined
+                name = 'Mycroft'
```

Comment:
Can we do `name = name if name else 'Mycroft'` so we can potentially keep the custom name if supplied?

Reply:
The inspect path does only occur when the custom name isn't supplied as I'm reading it?

'Mycroft' would only be used if the custom name is omitted and it tries to use inspect to generate the name and that inspect call fails.

Tested it out and the skills self.log.info comes out with skill names as expected.

Reply:
:+1: Thanks.